file/writable_file_writer.h

```diff
@@ -114,6 +114,17 @@ class WritableFileWriter {
     }
     info.status.PermitUncheckedError();
   }
+
+  void NotifyOnIOError(const IOStatus& io_status, const std::string& operation,
+                       const std::string& file_path, size_t length = 0,
+                       uint64_t offset = 0) {
+    IOErrorInfo io_error_info(io_status, operation, file_path, length, offset);
+
+    for (auto& listener : listeners_) {
+      listener->OnIOError(io_error_info);
+      // io_status->PermitUncheckedError();
```

Comment:
I think outside of the for block you probably need to do a io_error_info.io_status.PermitUncheckedError.

Also, should the input io_status have its checked status changed (if it was not checked before, should it be left unchecked after this call)?  It does not seem like this method should change the check status as you do not know if it did anything.

Reply:
Sure. I am adding the unit tests currently and will update the status checking.